se_ompl/include/se/ompl/collision_checker_voxel.hpp

```diff
@@ -81,8 +95,126 @@ const Planning_Configuration &planning_config
+}
+
+template<typename FieldType>
+bool CollisionCheckerV<FieldType>::isVoxelFree(const Eigen::Vector3i &point_v) const {
+
+  se::Node<FieldType> *node = nullptr;
+  se::VoxelBlock<FieldType> *block = nullptr;
+  bool is_voxel_block;
+  if (point_v.x() < 0 || point_v.y() < 0 || point_v.z() < 0 || point_v.x() > octree_ptr_->size()
+      || point_v.y() > octree_ptr_->size() || point_v.z() > octree_ptr_->size()) {
+    return false;
+  }
+  octree_ptr_->fetch_octant(point_v.x(), point_v.y(), point_v.z(), node, is_voxel_block);
+  if (is_voxel_block) {
+    block = static_cast<se::VoxelBlock<FieldType> *> (node);
+    if (block->data(point_v).x <= THRESH_FREE_LOG) {
+      // DLOG(INFO) << "free at "
+      // << (point_v.cast<float>() * voxel_dim_).format(InLine) << " state "
+      // << block->data(point_v).st << " prob "<< block->data(point_v).x ;
+
+      return true;
+    } else {
+      // DLOG(INFO)  << "collision at " << point_v.format(InLine) << " state "<<  block->data(point_v).st <<
+      // " prob " << block->data(point_v).x ;
+      return false;
+    }
+
+  } else {
+    const unsigned int
+        id = se::child_id(node->code_, octree_ptr_->leaf_level(), octree_ptr_->max_level());
+    const auto &data = node->parent()->value_[id];
+    if (data.x <= THRESH_FREE_LOG) {
+      // const Eigen::Vector3i pos = se::keyops::`decode(node->code_);
+      // LOG(INFO) << "collision at node "
+      // << (pos.cast<float>() * voxel_dim_).format(InLine) << std::endl;
+      return true;
+    } else
+      return false;
+  }
+
+}
+
+template<typename FieldType>
+bool CollisionCheckerV<FieldType>::isNodeFree(const key_t morton_code, const int node_level) const {
+  se::Node<FieldType> *node = nullptr;
+  se::VoxelBlock<FieldType> *block = nullptr;
+
+  if (node_level == octree_ptr_->leaf_level()) {
+    block = octree_ptr_->fetch(morton_code);
+    // LOG(INFO) << "VB morton "<< morton_code<< " prob " << block->data(VoxelBlock<OFusion>::buff_size - 1).x <<
+    // " state " << block->data(VoxelBlock<OFusion>::buff_size - 1).st;
+    if (block->data(VoxelBlock<OFusion>::buff_size - 1).x <= THRESH_FREE_LOG) {
+      return true;
+    } else {
+      return false;
+    }
+
+  } else {
+    const Eigen::Vector3i coord = se::keyops::decode(morton_code);
+    node = octree_ptr_->fetch_octant(coord.x(), coord.y(), coord.z(), node_level_);
+    const unsigned int id = se::child_id(morton_code, node_level_, octree_ptr_->max_level());
+    const auto &data = node->parent()->value_[id];
+    // LOG(INFO) << "Node morton " << morton_code << " level "<< node_level<<" prob "<< data.x << " state " << data.st;
+    if (data.x <= THRESH_FREE_LOG) {
+      return true;
+    } else {
+      return false;
+    }
+  }
+
+}
+
+template<typename FieldType>
+VecVec3i CollisionCheckerV<FieldType>::checkPointsAtNodeLevel( mapvec3i &node_list,
+                                                              const int node_level) const {
+
+  VecVec3i points;
```

Comment:
This contains the coordinates of the nodes that are occupied, right? In that case it should be renamed to something like occupied_nodes.